internal/gcsx/random_reader.go

```diff
@@ -256,14 +255,15 @@ func (rr *randomReader) ReadAt(
 
 	// Note: If we are reading the file for the first time and read type is sequential
 	// then the file cache behavior is write-through i.e. data is first read from
-	// GCS, cached in file and then served from that file. Also, the cacheHit is
-	// true in that case.
+	// GCS, cached in file and then served from that file. But the cacheHit is
+	// false in that case.
 	n, cacheHit, err = rr.tryReadingFromFileCache(ctx, p, offset)
 	if err != nil {
 		err = fmt.Errorf("ReadAt: while reading from cache: %v", err)
 		return
 	}
-	if cacheHit {
+	// data was served from cache.
```

Comment:
@raj-prince Could you please review this check ?

Reply:
Check looks good to me.

**One more option:**
We can make this second level check, less chances to hit n < len(p).
if cacheHit  then return 
else { 
if n == len(p) || (...) then return 
}